validator/sawtooth_validator/journal/chain.py

```diff
@@ -43,161 +54,246 @@ def __init__(self, consensus,
         self._executor = executor
         self._squash_handler = squash_handler
 
-    @property
-    def new_block(self):
-        return self._new_block
+        self._result = {
+            'new_block': new_block,
+            'chain_head': chain_head,
+            'new_chain': [],
+            'cur_chain': [],
+            'committed_batches': [],
+            'uncommitted_batches': [],
+        }
 
-    @property
-    def chain_head(self):
-        return self._chain_head
+    def _is_block_complete(self, blkw):
```

Comment:
The completer will only pass block to the journal if they are deem complete. What is your reasoning for "re-checking" the completeness?